### PR TITLE
[CPM-89] create data model for plan

### PR DESCRIPTION
**Pull request by HongShi10**
added plan data model and tests

### CODE REVIEW

File: api/src/models/student.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ const uniqueValidator = require("mongoose-unique-validator");
 /**
  * @swagger
  *  components:
+ *   $ref: './course.js/#components'
```
Comment (daltonlim) [inline api/src/models/student.js:6]: Uh dont know how to comment on unchanged code but is the todo at the bottom of this file supposed to be left like that?

File: api/src/models/plan.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "ProgrammeDegree",
+        required: true,
+    },
+    startYear: {
+        type: String,
+        required: true,
+    },
+    endYear: {
+        type: String,
```
Comment (daltonlim) [inline api/src/models/plan.js:71]: Should this be a number?

Comment (HongShi10) [reply api/src/models/plan.js:71]: i feel like string could work here as well

File: api/src/models/student.js (hunk 1 above)
Comment (HongShi10) [reply api/src/models/student.js:6]: i will change this

File: api/src/models/plan.js (hunk 2 above)
Comment (daltonlim) [reply api/src/models/plan.js:71]: but courseAllocationSchema has year as number

Comment (HongShi10) [reply api/src/models/plan.js:71]: true i will change boss

**Review by daltonlim — APPROVED** [review]
Thanks For The Changes 💯 LGTM

**Review by Dinith1 — APPROVED** [review]
Noice LGTM boss 🐊

**Pull request merged**